jacodb-panda-dynamic/src/main/kotlin/org/jacodb/panda/dynamic/ets/graph/EtsCfg.kt

```diff
@@ -0,0 +1,140 @@
+            }
+        }
+        map
+    }
+
+    override fun throwers(node: EtsStmt): Set<EtsStmt> {
+        TODO("Current version of IR does not contain try catch blocks")
+    }
+
+    override fun catchers(node: EtsStmt): Set<EtsStmt> {
+        TODO("Current version of IR does not contain try catch blocks")
+    }
+
+    override val instructions: List<EtsStmt>
+        get() = stmts
+
+    override val entries: List<EtsStmt>
+        get() = listOf(stmts.first())
+
+    override val exits: List<EtsTerminatingStmt>
+        get() = instructions.filterIsInstance<EtsTerminatingStmt>()
+
+    override fun successors(node: EtsStmt): Set<EtsStmt> {
+        return successorMap[node]!!.toSet()
+    }
+
+    override fun predecessors(node: EtsStmt): Set<EtsStmt> {
+        return predecessorMap[node]!!
+    }
+}
+
+fun EtsCfg.findDominators(): GraphDominators<EtsStmt> {
+    return GraphDominators(this).also { it.find() }
+}
+
+fun EtsCfg.view(dotCmd: String, viewerCmd: String, viewCatchConnections: Boolean = false) {
+    Util.sh(arrayOf(viewerCmd, "file://${toFile(dotCmd, viewCatchConnections)}"))
+}
+
+fun EtsCfg.toFile(dotCmd: String, viewCatchConnections: Boolean = false, file: File? = null): Path {
+    Graph.setDefaultCmd(dotCmd)
+
+    val graph = Graph("etsCfg")
+
+    val nodes = mutableMapOf<EtsStmt, Node>()
+    for ((index, inst) in instructions.withIndex()) {
+        val node = Node("$index")
+            .setShape(Shape.box)
+            .setLabel(inst.toString().replace("\"", "\\\""))
+            .setFontSize(12.0)
+        nodes[inst] = node
+        graph.addNode(node)
+    }
+
+    graph.setBgColor(Color.X11.transparent)
+    graph.setFontSize(12.0)
+    graph.setFontName("Fira Mono")
+
+    for ((inst, node) in nodes) {
+        when (inst) {
+            is EtsIfStmt -> {
+                val successors = successors(inst).toList()
+                check(successors.size == 2)
+                graph.addEdge(
+                    Edge(node.name, nodes[successors[0]]!!.name)
+                        .also {
+                            it.setLabel("false")
+                        }
+                )
+                graph.addEdge(
+                    Edge(node.name, nodes[successors[1]]!!.name)
+                        .also {
+                            it.setLabel("true")
+                        }
+                )
+            }
+
+            else -> for (successor in successors(inst)) {
+                graph.addEdge(Edge(node.name, nodes[successor]!!.name))
+            }
+        }
+        if (viewCatchConnections) {
+            // for (catcher in catchers(inst)) {
+            //     graph.addEdge(Edge(node.name, nodes[catcher]!!.name).also {
+            //         // it.setLabel("catch ${catcher.throwable.type}")
+            //         it.setLabel("catch")
+            //         it.setStyle(Style.Edge.dashed)
+            //     })
+            // }
```

Comment:
Why? Probably, you have to somehow connect this code with the corresponding TODO